index.js

```diff
@@ -45,12 +46,8 @@ export default class ClearvisioAppointmentBooker {
     this.setupCustomerFields(options);
     this.setupApi(options);
     this.loadStore(options.storeCode)
-      .then(() => {
-        return Promise.all([
-          this.loadEyeExaminationProcesses(options),
-          this.loadCalendars()
-        ]);
-      })
+      .then(() => this.loadCalendars())
+      .then(() => this.loadEyeExaminationProcesses(options))
```

Comment:
Azt javasolnám, hogy a lekérés részt az tegyük vissza promise allba, tehát párhuzamosan várjunk a szerverre, hogy milyen eljárások és calendarok vannak. csak aztán ha meg van mindkettő akkor szűrjük az examination processes eredményét

(Ennek egyik megoldása h itt rá thenelni a Promise.all-ra és kicsorgatni a hálózati kommunikáció eredményét, a másik hogy mint ahogy az availableCalendarFilter.js is használva van megjelenítésnél kimatekozni mindig lájv. (értem h az potenciálisan felesleges újraszámolgatás.))